codalab/lib/interactive_session.py

```diff
@@ -142,30 +143,34 @@ def get_docker_path(sub_path):
             else:
                 volumes[get_docker_path(key)] = dependency_local_path
 
-        name = self._get_container_name()
-        # Start a container as a root user. Root (id = 0) is the default user within a container.
-        command = [
+        name: str = self._get_container_name()
+        container_working_directory: str = f'{os.path.sep}{self._session_uuid}'
+
+        # Start a container as a non-root user
+        command: List[str] = [
             'docker run',
             '-it',
             f'--name {name}',
-            f'-w {os.path.sep}{self._session_uuid}',
-            # TODO: figure out a way to run as a non-root user and be able to save the bash_history on any machine
-            # '-u 1',
+            f'-w {container_working_directory}',
+            f'-e HOME={container_working_directory}',
+            f'-e HISTFILE={InteractiveSession._BASH_HISTORY_CONTAINER_PATH}',
+            '-e PROMPT_COMMAND="history -a"',
```

Comment:
Can you explain your approach here? Why is it needed to set PROMPT_COMMAND?

Reply:
This environment variable will ensure the history always gets appended to `$HISTFILE`.

Reply:
Ok, so does this happen by default for root users without needing to set PROMPT_COMMAND?

Reply:
I don't think was an issue before. The bash_history would get updated properly.